### PR TITLE
Cross-platform support on path

### LINKED ISSUE #62: Cross-platform support on path
On windows, running
```
oras push localhost:5000/oras:win dir\hello.txt
```
results in 
```json
{
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.oci.image.config.v1+json",
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "size": 2
  },
  "layers": [
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "digest": "sha256:0d74c782a0fd750336d9703eb9995985255bb5dd383c28d1828a6379a5418e4e",
      "size": 14,
      "annotations": {
        "org.opencontainers.image.title": "dir\\hello.txt"
      }
    }
  ]
}
```
Pulling that image under Linux/Unix will result in
```
-rw-rw-rw- 1 shizh shizh 14 Apr 11 18:12 'dir\hello.txt'
```
All paths should be converted to slash-seperated.

### PR DESCRIPTION
**Pull request by shizhMSFT**
Better cross-platform support on Windows.

With this patch,
- Windows path are automatically converted to "/" except absolute paths. For instances,
  - `foo\bar\hello.txt` --> `foo/bar/hello.txt`
  - `C:\hello.txt` --> `C:\hello.txt` (no change)
- Allow file reference with Windows absolute paths. For instance,
  - `C:\hello.txt`
  - `C:\hello.txt:vnd.your.media.type`

Fixes #62, #63

### CODE REVIEW

File: cmd/oras/push_windows.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package main
```
Comment (jdolitsky) [inline cmd/oras/push_windows.go]: do we need a `// +build` here too?

Comment (shizhMSFT) [reply cmd/oras/push_windows.go]: No need for that. Just like `*_test.go`

See https://golang.org/pkg/go/build/#hdr-Build_Constraints

**Comment by shizhMSFT** [comment]
> What is the override flag here to indicate that you are allowing breakout from current directory?

I think you are commeting on #66

**Pull request merged**